app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java

```diff
@@ -0,0 +1,427 @@
+      // Prepare for input in case of empty/new note
+      addNoteEditText.requestFocus();
+      showKeyboard();
+    }
+  }
+
+  public void showKeyboard(){
+    InputMethodManager inputMethodManager = (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+    inputMethodManager.toggleSoftInput(InputMethodManager.SHOW_FORCED, 0);
+  }
+
+  public void closeKeyboard(){
+    InputMethodManager inputMethodManager = (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+    inputMethodManager.toggleSoftInput(InputMethodManager.HIDE_IMPLICIT_ONLY, 0);
+  }
+
+  private void saveNote(String noteText) {
+
+    /* String content of the EditText, given by noteText, is saved into the text file given by:
+     *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+     * */
+
+    if(isExternalStorageWritable()) {
+
+      if(ContextCompat.checkSelfPermission(getContext(), Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+        Log.d(TAG, "WRITE_EXTERNAL_STORAGE permission not granted");
+        Toast.makeText(getContext(), getString(R.string.note_save_unsuccessful), Toast.LENGTH_LONG);
+        return;
+      }
+
+      File notesFolder = new File(Environment.getExternalStorageDirectory() + "/Kiwix/Notes/" + zimFileTitle);
+      boolean folderExists = true;
+
+      if(!notesFolder.exists()) {
+        // Try creating folder if it doesn't exist
+        folderExists = notesFolder.mkdirs();
+      }
+
+      if(folderExists) {
+        File noteFile = new File(notesFolder.getAbsolutePath(), articleTitle + ".txt");
+
+        // Save note text-file code:
+        try {
+          FileOutputStream fileOutputStream = new FileOutputStream(noteFile);
+          fileOutputStream.write(noteText.getBytes());
+          fileOutputStream.close();
+          Toast.makeText(getContext(), getActivity().getString(R.string.note_save_successful), Toast.LENGTH_SHORT).show();
+          noteEdited = false; // As no unsaved changes remain
+
+        } catch (IOException e) {
+          e.printStackTrace();
+          Toast.makeText(getContext(), getActivity().getString(R.string.note_save_unsuccessful), Toast.LENGTH_LONG);
+        }
+
+      } else {
+        Toast.makeText(getContext(), getActivity().getString(R.string.note_save_unsuccessful), Toast.LENGTH_LONG);
+        Log.d(TAG, "Required folder doesn't exist");
+      }
+    }
+    else {
+      Toast.makeText(getContext(), getString(R.string.note_save_error_storage_not_writable), Toast.LENGTH_LONG).show();
+    }
+
+  }
+
+  private void displayNote() {
+
+    /* String content of the note text file given at:
+     *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+     * is displayed in the EditText field (note content area)
+     * */
+
+    File noteFile = new File(Environment.getExternalStorageDirectory() + "/Kiwix/Notes/" + zimFileTitle + "/" + articleTitle + ".txt");
```

Comment:
extract "Environment.getExternalStorageDirectory() + "/Kiwix/Notes/"" to a private static field named NOTES_DIRECTORY probably, adds more meaning and prevents you from repeating yourself

Reply:
Added a [constant string](https://github.com/kiwix/kiwix-android/pull/1198/commits/873eb2bce86c76b11962367701cb66bdf66b9f5e#diff-62841bd47f525ee302268e9b480bfb8aR143) & refactored [everywhere](https://github.com/kiwix/kiwix-android/commit/873eb2bce86c76b11962367701cb66bdf66b9f5e)